src/main/webapp/components/Documents.js

```diff
@@ -156,6 +150,8 @@ export const Documents = hh(class Documents extends Component {
         isRendered: this.props.isConsentGroup === true
       }, [
           DataUseLetter({
+            showSpinner: this.props.showSpinner,
+            hideSpinner: this.props.hideSpinner,
```

Comment:
Just commenting here so I'm clear on the ownership. In both `AddDocumentDialog` and `DataUseLetter`, we are giving them the same spinner from the parent component. If true, is that intentional or should they have their own spinner?

Reply:
Additional question, if they are the same, will an action in `AddDocumentDialog` have any effect on the spinner behavior in `DataUseLetter`?

Reply:
Current approach is to have a single spinner, wrapping top component. So, all components in Routes.js (Switch) will share same spinner. Traversing the component tree, we may or may not propagate same spinner. Or we could wrap an inner component with a new spinner. In this particular case, AddDocumentDialog and DataUseLetter will share same spinner (top spinner, the only one now) showing or hiding in different situations . New spinner component let us to wrap any component in it own spinner. We may need to review component tree and spinners, but I think it may be another story.

Reply:
I'm a little confused by this. The intention should have been to let every component have it's own spinner. Looking back at the ticket, I think the original intent was misunderstood. The idea behind introducing an HOC here was not to make the spinner global, but instead was to distribute the spinner handling down to each component that needed it. 

> Current approach is to have a single spinner, wrapping top component

☝️ That approach replicates what we had before, just a different implementation.

Reply:
Some links to previous comments that explain the problems with the global approach:
* https://github.com/broadinstitute/orsp-pub/pull/226#pullrequestreview-271412490
* https://github.com/broadinstitute/orsp-pub/pull/236#discussion_r313615963